src/Libraries/RevitNodes/Filter/FilterRule.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using Autodesk.Revit.DB;
+using DynamoServices;
+using Autodesk.DesignScript.Runtime;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using System.Collections.Generic;
+
+namespace Revit.Filter
+{
+    /// <summary>
+    /// Revit Filter Rule
+    /// </summary>
+    public class FilterRule
+    {
+
+        #region Internal Properties
+
+        /// <summary>
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.FilterRule InternalFilterRule
+        { 
+            get; set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        /// <param name="internalFilterRule"></param>
+        internal FilterRule(Autodesk.Revit.DB.FilterRule internalFilterRule)
+        {
+            this.InternalFilterRule = internalFilterRule;
+        }
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// FilterType Enumeration
+        /// </summary>
+        [IsVisibleInDynamoLibrary(false)]
+        public enum FilterType
+        {
+            BeginsWith,
+            Contains,
+            EndsWith,
+            Equals,
+            Greater,
+            Less,
+            GeaterOrEqual,
+            LessOrEqual,
+            NotBeginsWith,
+            NotContains,
+            NotEndsWith,
+            NotEquals,
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Create a new Filter Rule
+        /// </summary>
+        /// <param name="type">Filter Rule Type</param>
+        /// <param name="value">Value to check</param>
+        /// <param name="parameter">Parameter to filter</param>
+        /// <returns></returns>
+        public static FilterRule ByRuleType(string typename, object value, Elements.Parameter parameter)
```

Comment:
@sharadkjaiswal @mjkkirschner can you cast a string into an enum in Dynamo? Because for the rebar package for example most users are constructing their enum values as strings. Maybe we should add a enum cast node allowing the users to get their enum value from strings?